From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Etil <81570777+etil2jz@users.noreply.github.com>
Date: Thu, 2 Dec 2021 16:06:46 +0100
Subject: [PATCH] Mirai Configuration


diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index e0f69edf603c2ec99bc92b16b18912272cc41bd9..5e951eba9c4865a4727c4c0e58783b0ece357ac0 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -1983,6 +1983,13 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
         }
         // Purpur end
 
+        // Mirai start
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getMiraiConfig() {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+        // Mirai end
+
         /**
          * Sends the component to the player
          *
