From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Etil <81570777+etil2jz@users.noreply.github.com>
Date: Thu, 2 Dec 2021 16:06:46 +0100
Subject: [PATCH] Mirai Configuration


diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index a88c68c089805dcec131f28c9b3471ba34838b22..0d62241f872b97b40d85bed6b0630f6814f1d983 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2091,6 +2091,13 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
         }
 		// Leaves end - Not supported yet too
 
+        // Mirai start
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getMiraiConfig() {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+        // Mirai end
+
         /**
          * Sends the component to the player
          *
