From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Etil <81570777+etil2jz@users.noreply.github.com>
Date: Thu, 2 Dec 2021 16:06:46 +0100
Subject: [PATCH] Mirai Configuration


diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index d0c634629aa0b6bac0da93655dd86ad3aea0ce30..38457ee73147f28c5ac3097090680a8b0875fb0d 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2190,6 +2190,13 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
         }
         // Paper end
 
+        // Mirai start
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getMiraiConfig() {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+        // Mirai end
+
         /**
          * Sends the component to the player
          *
