From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Dreeam <61569423+Dreeam-qwq@users.noreply.github.com>
Date: Mon, 22 May 2023 08:06:21 +0800
Subject: [PATCH] Mirai: Add Higher Java Versions Support for SIMD


diff --git a/src/main/java/gg/pufferfish/pufferfish/simd/SIMDChecker.java b/src/main/java/gg/pufferfish/pufferfish/simd/SIMDChecker.java
index ab5fea0b03224bf249352ce340e94704ff713345..3f19714cf810e4b0a1bc0e903bc9f1907cce1741 100644
--- a/src/main/java/gg/pufferfish/pufferfish/simd/SIMDChecker.java
+++ b/src/main/java/gg/pufferfish/pufferfish/simd/SIMDChecker.java
@@ -15,7 +15,7 @@ public class SIMDChecker {
 	@Deprecated
 	public static boolean canEnable(Logger logger) {
 		try {
-			if (SIMDDetection.getJavaVersion() != 17 && SIMDDetection.getJavaVersion() != 18 && SIMDDetection.getJavaVersion() != 19) {
+			if (SIMDDetection.getJavaVersion() < 17) { // Mirai - Add Higher Java Versions Support for SIMD
 				return false;
 			} else {
 				SIMDDetection.testRun = true;
