From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Etil <81570777+etil2jz@users.noreply.github.com>
Date: Thu, 9 Dec 2021 15:39:29 +0100
Subject: [PATCH] Configurable max bees in hive


diff --git a/src/main/java/dev/etil/mirai/MiraiConfig.java b/src/main/java/dev/etil/mirai/MiraiConfig.java
index af9303f211c340473b6b236216ff19d3417e4b28..15973a524157b3301960365c7e9dd09b6bbeec91 100644
--- a/src/main/java/dev/etil/mirai/MiraiConfig.java
+++ b/src/main/java/dev/etil/mirai/MiraiConfig.java
@@ -136,4 +136,11 @@ public class MiraiConfig {
         logPlayerLoginLoc = getBoolean("log-player-login-location", true,
             "Whether or not player login location should be written in logs.");
     }
+
+    public static int maxBees;
+    private static void maximumBees() {
+        maxBees = getInt("max-bees-in-hive", 3,
+            "This value defines how many bees can fit in a hive.",
+            "Allowing more bees in a hive can reduce the number of ticking hives.");
+    }
 }
diff --git a/src/main/java/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java b/src/main/java/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
index dfd364ac4b7551a13c4c6c100b5e62c0dfb10595..f3299759067698f73d296268c1a5277c5644a2ea 100644
--- a/src/main/java/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
+++ b/src/main/java/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
@@ -43,7 +43,7 @@ public class BeehiveBlockEntity extends BlockEntity {
     private final List<BeehiveBlockEntity.BeeData> stored = Lists.newArrayList();
     @Nullable
     public BlockPos savedFlowerPos;
-    public int maxBees = 3; // CraftBukkit - allow setting max amount of bees a hive can hold
+    public int maxBees = dev.etil.mirai.MiraiConfig.maxBees; // CraftBukkit - allow setting max amount of bees a hive can hold // Mirai - configurable max bees in hive
 
     public BeehiveBlockEntity(BlockPos pos, BlockState state) {
         super(BlockEntityType.BEEHIVE, pos, state);
