From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: etil2jz <81570777+etil2jz@users.noreply.github.com>
Date: Fri, 4 Mar 2022 19:57:07 +0100
Subject: [PATCH] Skip cloning loot parameters


diff --git a/src/main/java/net/minecraft/advancements/Advancement.java b/src/main/java/net/minecraft/advancements/Advancement.java
index 81359be381fc9bcb56a9cc83e70a6afa6e838c6a..4a3e86d9b99ba26467c81ddc66f709cdbfd6d55f 100644
--- a/src/main/java/net/minecraft/advancements/Advancement.java
+++ b/src/main/java/net/minecraft/advancements/Advancement.java
@@ -46,7 +46,7 @@ public class Advancement {
     public Advancement(ResourceLocation id, @Nullable Advancement parent, @Nullable DisplayInfo display, AdvancementRewards rewards, Map<String, Criterion> criteria, String[][] requirements, boolean sendsTelemetryEvent) {
         this.id = id;
         this.display = display;
-        this.criteria = ImmutableMap.copyOf(criteria);
+        this.criteria = java.util.Collections.unmodifiableMap(criteria); // Mirai - Skip cloning loot parameters
         this.parent = parent;
         this.rewards = rewards;
         this.requirements = requirements;
