From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: etil2jz <81570777+etil2jz@users.noreply.github.com>
Date: Tue, 2 Aug 2022 14:48:12 +0200
Subject: [PATCH] Configurable chat message signatures


diff --git a/src/main/java/dev/etil/mirai/MiraiConfig.java b/src/main/java/dev/etil/mirai/MiraiConfig.java
index 838c2ece553c2ccb6ebfa2fe95e2998300d73ffd..df958030e975c3910d8f875f0309326a692fc7c0 100644
--- a/src/main/java/dev/etil/mirai/MiraiConfig.java
+++ b/src/main/java/dev/etil/mirai/MiraiConfig.java
@@ -186,4 +186,12 @@ public class MiraiConfig {
         optimizedHypot = getBoolean("optimize-math-hypot", true,
             "Whether or not Math.hypot should be replaced by a faster version.");
     }
+
+    public static boolean chatMessageSignatures;
+    private static void chatreport() {
+        chatMessageSignatures = getBoolean("disable-chat-reports", false,
+            "Whether or not players should be able to report chat messages.",
+            "It also disables the popup when joining a server without",
+            "'secure chat', such as offline-mode servers.");
+    }
 }
diff --git a/src/main/java/net/minecraft/network/protocol/status/ServerStatus.java b/src/main/java/net/minecraft/network/protocol/status/ServerStatus.java
index 11183b4a39b9897e1b1067eb2a3556d2b1d90f52..df109dd1b2d252f82096d02e57596fdd4076771f 100644
--- a/src/main/java/net/minecraft/network/protocol/status/ServerStatus.java
+++ b/src/main/java/net/minecraft/network/protocol/status/ServerStatus.java
@@ -78,6 +78,7 @@ public class ServerStatus {
     }
 
     public boolean enforcesSecureChat() {
+        if (dev.etil.mirai.MiraiConfig.chatMessageSignatures) return true; // Mirai - configurable chat message signatures
         return this.enforcesSecureChat;
     }
 
diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index 6f3da8a778127ff273aea6cc6465485f21d7bbbb..b3b6a33bba5aa43fc0e543ee68d68f7ddb4b45ff 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -683,6 +683,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
 
     @Override
     public boolean enforceSecureProfile() {
+        if (dev.etil.mirai.MiraiConfig.chatMessageSignatures) return false; // Mirai - configurable chat message signatures
         return this.getProperties().enforceSecureProfile && this.getProperties().onlineMode;
     }
 
diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index acecb7cf4e91757dd54566b95995af6e13eed78b..ce59b4edbb1ab512c933383ff258c3e172189919 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -2240,10 +2240,34 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Tic
 
     @Override
     public void send(Packet<?> packet) {
+        // Mirai start - configurable chat message signatures
+        if (dev.etil.mirai.MiraiConfig.chatMessageSignatures) {
+            if (packet instanceof net.minecraft.network.protocol.game.ClientboundPlayerChatHeaderPacket) {
+                return;
+            } else if (packet instanceof net.minecraft.network.protocol.game.ClientboundPlayerChatPacket chat) {
+                packet = new ClientboundSystemChatPacket(chat.chatType().resolve(this.player.level.registryAccess())
+                    .get().decorate(chat.message().serverContent()), false);
+
+
+                this.send(packet);
+                return;
+            }
+        }
+        // Mirai end
         this.send(packet, (PacketSendListener) null);
     }
 
     public void send(Packet<?> packet, @Nullable PacketSendListener callbacks) {
+        // Mirai start - configurable chat message signatures
+        if (dev.etil.mirai.MiraiConfig.chatMessageSignatures) {
+            if (packet instanceof net.minecraft.network.protocol.game.ClientboundPlayerChatHeaderPacket) {
+                return;
+            } else if (packet instanceof net.minecraft.network.protocol.game.ClientboundPlayerChatPacket chat && callbacks != null) {
+                this.send(chat);
+                return;
+            }
+        }
+        // Mirai end
         // CraftBukkit start
         if (packet == null || this.processedDisconnect) { // Spigot
             return;
diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index f93f72fb6d2258c06bb8661f8624567e47c8766c..6376c1bc4b1d1e8cd0acede96d68505d06ebd4e2 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -1530,6 +1530,7 @@ public abstract class PlayerList {
     }
 
     public boolean verifyChatTrusted(PlayerChatMessage message, ChatSender profile) { // Paper - private -> public
+        if (dev.etil.mirai.MiraiConfig.chatMessageSignatures) return true; // Mirai - configurable chat message signatures
         return !message.hasExpiredServer(Instant.now()) && message.verify(profile);
     }
 
diff --git a/src/main/java/net/minecraft/world/entity/player/Player.java b/src/main/java/net/minecraft/world/entity/player/Player.java
index ede7a3656d9c19fcaefcc0936a280271cdb966f8..7e20e25060892dedb21a378bdb9ea98c50a702f4 100644
--- a/src/main/java/net/minecraft/world/entity/player/Player.java
+++ b/src/main/java/net/minecraft/world/entity/player/Player.java
@@ -1566,6 +1566,7 @@ public abstract class Player extends LivingEntity {
 
     @Nullable
     public ProfilePublicKey getProfilePublicKey() {
+        if (dev.etil.mirai.MiraiConfig.chatMessageSignatures) return null; // Mirai - configurable chat message signatures
         return this.profilePublicKey;
     }
 
