From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rafaelflromao <12960698+rafaelflromao@users.noreply.github.com>
Date: Mon, 12 Jun 2023 06:01:40 +0100
Subject: [PATCH] Sync event calls on async threads


diff --git a/src/main/java/dev/etil/mirai/MiraiConfig.java b/src/main/java/dev/etil/mirai/MiraiConfig.java
index 1cdab691bef60250fb4f3ff35d270caa99e3af6b..5d1dc477e1ac07a1fefdc182bbb9ee0c2dde828f 100644
--- a/src/main/java/dev/etil/mirai/MiraiConfig.java
+++ b/src/main/java/dev/etil/mirai/MiraiConfig.java
@@ -289,4 +289,16 @@ public class MiraiConfig {
             enableAsyncEntityTracker = temp;
         }
     }
+
+    public static boolean enableSyncEventCallsOnAsyncThreads;
+    public static boolean enableSyncEventCallsOnAsyncThreadsInitialized;
+    private static void syncEventCallsOnAsyncThreads() {
+        boolean temp = getBoolean("enable-sync-event-calls-on-async-threads", true,
+            "Whether or not sync event calls on async threads should be enabled. (If async entity tracker is enabled, this is enabled.)",
+            "You may encounter issues with plugins.");
+        if (!enableSyncEventCallsOnAsyncThreadsInitialized) {
+            enableSyncEventCallsOnAsyncThreadsInitialized = true;
+            enableSyncEventCallsOnAsyncThreads = temp;
+        }
+    }
 }
diff --git a/src/main/java/io/papermc/paper/plugin/manager/PaperEventManager.java b/src/main/java/io/papermc/paper/plugin/manager/PaperEventManager.java
index 32bd9b4e4f77598806962cb6a91c0f04455a7133..21844207d3acf8b86367db95a23f5cbc57fb44c5 100644
--- a/src/main/java/io/papermc/paper/plugin/manager/PaperEventManager.java
+++ b/src/main/java/io/papermc/paper/plugin/manager/PaperEventManager.java
@@ -41,7 +41,7 @@ class PaperEventManager {
             throw new IllegalStateException(event.getEventName() + " may only be triggered asynchronously.");
         } else if (!event.isAsynchronous() && !this.server.isPrimaryThread() && !this.server.isStopping()) {
             // Mirai start
-            if (dev.etil.mirai.MiraiConfig.enableAsyncEntityTracker) {
+            if (dev.etil.mirai.MiraiConfig.enableAsyncEntityTracker || dev.etil.mirai.MiraiConfig.enableSyncEventCallsOnAsyncThreads) {
                 MinecraftServer.getServer().executeBlocking(event::callEvent);
                 return;
             } else {
